sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/annotation/ForbidAnnotationElementValueCheck.java

```diff
@@ -0,0 +1,386 @@
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.annotation;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.base.CharMatcher;
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Forbids specific
+ * <a href= 'https://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.6.1'>element
+ * value</a> for specific annotation. You can configure this check using following options:
+ * </p>
+ * <ul>
+ * <li>Annotation name</li>
+ * <li>Annotation element name</li>
+ * <li>Forbidden annotation element value pattern</li>
+ * </ul>
+ * Example of usage:<br>
+ * <p>
+ * Here is XML configs and according code samples needed to forbid
+ * </p>
+ * <p>
+ * Config
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="ForbidAnnotationElementValue"&gt;
+ * &lt;property name="annotationName" value="Bean"/&gt;
+ * &lt;property name="elementName" value="name"/&gt;
+ * &lt;property name="forbiddenElementValueRegexp" value="AnnotationConfigUtils\..*"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code
+ * </p>
+ *
+ * <pre>
+ *  &#64;Bean(name = AnnotationConfigUtils.SCHEDULED_ANNOTATION_PROCESSOR_BEAN_NAME)
+ * </pre>
+ * <p>
+ * If you want to forbid a
+ * <a href= 'https://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.7.3'>single-element
+ * </a> annotation element value elementName option should be specified as "value".
+ * </p>
+ * <p>
+ * Config
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="ForbidAnnotationElementValue"&gt;
+ * &lt;property name="annotationName" value="SuppressWarnings"/&gt;
+ * &lt;property name="elementName" value="value"/&gt;
+ * &lt;property name="forbiddenElementValueRegexp" value="unchecked"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code
+ * </p>
+ *
+ * <pre>
+ * &#64;SuppressWarnings("unchecked")
+ * </pre>
+ * <p>
+ * To forbid any array-valued element, frobiddenElementValueRegexp option should be: "\{.*\}".
+ * </p>
+ * <p>
+ * Config
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="ForbidAnnotationElementValue"&gt;
+ * &lt;property name="annotationName" value="Endorsers"/&gt;
+ * &lt;property name="elementName" value="value"/&gt;
+ * &lt;property name="forbiddenElementValueRegexp" value="}\{.*\}"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>
+ * Code
+ * </p>
+ *
+ * <pre>
+ * &#64;Endorsers({"Children", "Unscrupulous dentists"})<br>

```

Comment:
example should be close to real life, if nothing come to mind - make example base on SuppressWarning annotation.